### PR TITLE
EventList Front-End

### PR DESCRIPTION
**Pull request by Lhyejin**
#37 #40
- View events received from server in Landing page and Eventlist page
- Apply material-ui to the events filter button
- Implement event filter function
- Implement designed event list page using material-ui

### CODE REVIEW

File: web/src/components/event-cards-query.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class EventCardsQuery extends Component {
+  constructor(props) {
+    /**
+     * Props
+     * pageSize {int} : Number of event card that bring from server once
+     * selectedTagIds {Array<Int>} : Array of selected tag id
+     * isRecommended {boolean} : whether recommend or not (current not use)
+     * after {int} :
+     * onCreate : HandlerCurrentCursor function of parent(eventlist.js)
+     * */
```
Comment (lsh9034) [inline web/src/components/event-cards-query.js]: Could you write JSDoc like this?
/**
  * @param {int} pageSize: decription
...
...
* */

File: web/src/components/event-cards.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                        src={require('./images/' + (cardIndex + 1) + '.jpg')}
+                />
+              }
+              title={this.props.events[cardIndex].title}
+              subheader="September 14, 2016"
+            />
+            {/* Content section of Card */}
+            <CardContent>
+              { /* 장소:{this.props.events[cardIndex].schedule[0].address}*/}
+              <Typography> 장소 </Typography>
```
Comment (lsh9034) [inline web/src/components/event-cards.js]: Could you change korean to english?

File: web/src/components/eventlist-filters.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  }
+});
+
+class EventlistFilters extends Component {
+  constructor(props) {
+    /**
+     * Props
+     * filterNames {Array<object tag>} : Entire tag Information
+     * onCreate : HandlerTagName function of parent(eventlist.js)
+     */
```
Comment (lsh9034) [inline web/src/components/eventlist-filters.js]: Change this part to my first comment way

File: web/src/components/eventlist-filters.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      filterClicked: filterClicked,
+    };
+  }
+
+  /**
+   * Handler function, When Tag Button is clicked
+   * Reverse filterClicked state clicked
+   * Send filterClicked to parent component using onCreate function
+   */
+  HandlerTagButton = (event, value) => {
```
Comment (lsh9034) [inline web/src/components/eventlist-filters.js]: Add @param to explain event, value variable

File: web/src/components/eventlist-filters.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  /**
+   * Create tag button list
+   * @return {Array<ToggleButton>} tagButtonArray
+   */
+  ViewTagButtons = () => {
+    const { classes } = this.props;
+    const tagButtonArray = [];
+
+    for (let tagIndex =0; tagIndex < this.props.filterNames.length; ++tagIndex)
```
Comment (lsh9034) [inline web/src/components/eventlist-filters.js]: let tagIndex = 0

File: web/src/pages/eventlist.js [hunk 6]
```diff
@@ -1,75 +1,100 @@
...
 
+  constructor() {
+    super();
+    this.state = {
+      isTagNames : false,
+      allTags : [],
+      selectedTagIds : [],
+      eventListPageSize: 9,
+      currentCursor: 0
+    }
```
Comment (lsh9034) [inline web/src/pages/eventlist.js]: don't leave space between variable name and ':' 
change like this
variable: value

File: web/src/pages/eventlist.js [hunk 7]
```diff
@@ -1,75 +1,100 @@
...
+   * @param cursor {int} : Endpoint of current viewed EventCard
+   */
+  HandlerCurrentCursor = (cursor) => {
+    this.setState({currentCursor: cursor})
   };
 
-  // Will add Skeleton component during loading
+  /**
+   * Set selectedTagIds state using selectedTags parameter
+   * @param selectedTags {Array<boolean>} : selected tag array
```
Comment (lsh9034) [inline web/src/pages/eventlist.js]: switch position variable name and type

File: web/src/pages/eventlist.js [hunk 8]
```diff
@@ -1,75 +1,100 @@
...
-        if (loading) return "Loading...";
-        if (error) return `Error! ${error.message}`;
-
-        //will add Skeleton component during loading
-        return(<EventCards>{data.events.events}</EventCards>)
-      }}
-    </Query>);
+    const selectedTagIds = [];
+    // Put selected tag id in selctedTagIds array
+    for(let tagIndex=0; tagIndex< this.state.allTags.length; ++tagIndex)
```
Comment (lsh9034) [inline web/src/pages/eventlist.js]: change let tagIndex = 0; tagIndex < this.state.allTags.length;

File: web/src/components/event-cards.js [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+      cards[cards.length - 1].push(
+        <Grid item xs={12} sm={6} md={4}>
+          <Card className={classes.card}
+                key={this.props.events[cardIndex].id}
+                value={this.props.events[cardIndex].id}
+                onClick={
+                  () => this.CardClicked(this.props.events[cardIndex].id)
+                }
+          >
+            {/* Image section of Card */}
```
Comment (shberry) [inline web/src/components/event-cards.js]: Indentation is off from here and below lines

File: web/src/components/event-cards.js [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+              title={this.props.events[cardIndex].title}
+              subheader="September 14, 2016"
+            />
+            {/* Content section of Card */}
+            <CardContent>
+              <Typography>
+                { /* {this.props.events[cardIndex].schedule[0].address}*/}
+              </Typography>
+              <Typography
+                className={classes.cardContent}>
```
Comment (shberry) [inline web/src/components/event-cards.js]: this can move to above line

File: web/src/components/eventlist-filters.js [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  /**
+   * Create tag button list
+   * @return {Array<ToggleButton>} tagButtonArray
+   */
+  ViewTagButtons = () => {
+    const { classes } = this.props;
+    const tagButtonArray = [];
+
+    for (let tagIndex = 0; tagIndex < this.props.filterNames.length; ++tagIndex)
+    {
```
Comment (shberry) [inline web/src/components/eventlist-filters.js]: move this to above line?

File: web/src/pages/eventlist.js [hunk 12]
```diff
@@ -1,75 +1,100 @@
...
+    this.state = {
+      isTagNames: false,
+      allTags: [],
+      selectedTagIds: [],
+      eventListPageSize: 9,
+      currentCursor: 0
+    }
   }
 
+  // // Query that bring Tag Names
```
Comment (shberry) [inline web/src/pages/eventlist.js]: remove this block?

File: web/src/pages/eventlist.js [hunk 13]
```diff
@@ -1,75 +1,100 @@
...
+            !this.state.isTagNames && <div></div>}
+              {/*<Query query={this.TAGNAMES_REQUEST_QUERY}*/}
+              {/*       onCompleted={data =>*/}
+              {/*       {this.setState({isTagNames: true,*/}
+              {/*         allTags: data.tagNames.tags});}}>*/}
+              {/*{({loading, error, data}) => {*/}
+              {/*  if (loading) return "Loading...";*/}
+              {/*  if (error) return `Error! ${error.message}`;*/}
+              {/*}}*/}
+              {/*</Query>}*/}
```
Comment (shberry) [inline web/src/pages/eventlist.js]: remove this block?

File: web/src/pages/eventlist.js [hunk 14]
```diff
@@ -1,75 +1,100 @@
...
+
+          <Grid container justify="space-between"
+                style={{padding: '2% 5% 0 5%'}}>
+            <Grid item xs md xl>
+              <EventlistFilters filterNames={this.state.allTags}
+                                onCreate={this.HandlerTagName}
+              />
```
Comment (shberry) [inline web/src/pages/eventlist.js]: move this to above line?

File: web/src/components/event-cards.js (hunk 9 above)
Comment (Lhyejin) [reply web/src/components/event-cards.js]: but from this comment to below lines above the card tag, they are in card tag. if Indentation remove, card tag and card content are placed same position.

**Pull request merged**